### PR TITLE
add GCE PD volume e2e tests

### PR DESCRIPTION
**Pull request by rootfs**
add GCE PD e2e volume test

@kubernetes/sig-storage

### CODE REVIEW

**Comment by saad-ali** [comment]
There are a bunch of tests for GCE PD and AWS EBS in [`test/e2e/pd.go`](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/pd.go) that do essentially the samething and have been battle tested (flakes worked out over time). So I do not think that this is really necessary.

**Comment by rootfs** [comment]
@saad-ali yes, `pd.go` has lots of test, but adding pd to `volumes.go` can exercise `securityContext` that is  not covered in `pd.go`

File: test/e2e/volumes.go [hunk 1]
```diff
@@ -340,6 +340,16 @@ func deleteCinderVolume(name string) error {
 	return err
 }
 
+func deletePDVolume(name string) error {
+	output, err := exec.Command("gcloud", "compute", "disks", "delete", name, "-q", "--zone", framework.TestContext.CloudConfig.Zone).CombinedOutput()
```
Comment (saad-ali) [inline test/e2e/volumes.go]: Use the GCE API, it is far more robust. See https://github.com/kubernetes/kubernetes/blob/master/test/e2e/pd.go#L323. Factor out the `createPD()` code from their to share if it makes sense.

File: test/e2e/volumes.go [hunk 2]
```diff
@@ -725,4 +735,53 @@ var _ = framework.KubeDescribe("Volumes [Feature:Volumes]", func() {
...
+	////////////////////////////////////////////////////////////////////////
+	// GCE PD
+	////////////////////////////////////////////////////////////////////////
+
+	// This test assumes that Google Cloud SDK is installed and
+	// client is authenticated
+
+	framework.KubeDescribe("PD", func() {
+		It("should be mountable", func() {
+			framework.SkipUnlessProviderIs("gce")
```
Comment (saad-ali) [inline test/e2e/volumes.go]: or `gke`

**Comment by rootfs** [comment]
@saad-ali feedback addressed. I have trouble bringing up gce cluster see #25198

**Comment by saad-ali** [comment]
Thanks Huamin
LGTM

**Comment by roberthbailey** [comment]
@rootfs can you squash your commits?

**Comment by rootfs** [comment]
@roberthbailey done, thanks!

**Comment by rootfs** [comment]
@roberthbailey oops, lgtm also squashed

**Pull request merged**